src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+
+interface IUseLocalStorageProps {
+  getLocalStorage: (key: string) => unknown | undefined
+  setLocalStorage: (key: string, value: unknown) => void
+  removeLocalStorage: (key: string) => void
+}
+
+const useLocalStorage = (): IUseLocalStorageProps => {
+  return React.useMemo(() => {
+    const getLocalStorage = (key: string) => {
+      const data = window.localStorage.getItem(key)
+
+      if (!data) return
+
+      return JSON.parse(data)
+    }
+
+    const setLocalStorage = (key: string, value: unknown) => {
+      const data = JSON.stringify(value)
+
+      return window.localStorage.setItem(key, data)
+    }
+
+    const removeLocalStorage = (key: string) => {
+      return window.localStorage.removeItem(key)
+    }
+
+    return {
+      getLocalStorage,
+      setLocalStorage,
+      removeLocalStorage
+    }
+  }, [])
+}
+
+export default useLocalStorage
```

Comment:
You're using `useMemo` not in a way it's supposed to be used, I guess the correct hook here is `useCallback` for each function.

I'm not saying it won't work, you can decide on what you think is better

```suggestion
const useLocalStorage = (): IUseLocalStorageProps => {
  const getLocalStorage = React.useCallback((key: string) => {
    const data = window.localStorage.getItem(key)

    return data ? JSON.parse(data) : undefined
  }, [])

  const setLocalStorage = React.useCallback((key: string, value: unknown) => {
    const data = JSON.stringify(value)

    return window.localStorage.setItem(key, data)
  }, [])

  const removeLocalStorage = React.useCallback((key: string) => {
    return window.localStorage.removeItem(key)
  }, [])

  return {
    getLocalStorage,
    setLocalStorage,
    removeLocalStorage
  }
}
```